docs/spec/fastsync/fastsync.md

```diff
@@ -0,0 +1,1164 @@
+terminates.  There is a constant *D >= 1* such that when *Fastsync*
+terminates, it outputs a list of all blocks from height *h* to some
+height *terminationHeight >= eh - D*.
+
+
+#### **[FS-SEQ-INV]**:
+Upon termination, the application state is the one that corresponds to
+the blockchain at height *terminationHeight*.
+
+> As the block of height *h + 1* is needed to verify the block of
+> height *h* we highlight th following clarification of the
+> termination height.
+
+
+#### **[FS-SEQ-HEIGHT]**: 
+The returned value *terminationHeight* is the height of the block with the largest
+height that could be verified. In order to do so, *Fastsync* needs the
+Commit of the block at height  *terminationHeight + 1* in the blockchain.
+
+
+# Part II - Protocol view
+
+## Environment/Assumptions/Incentives
+
+
+#### **[FS-A-NODE]**:
+We consider a node *FS* that performs *Fastsync*.
+
+#### **[FS-A-PEER-IDS]**:
+*FS* has access to a set *peerIDs* of IDs (public keys) of peers (full
+     nodes). During the execution of *Fastsync*, another protocol (outside
+     of this specification) may add new IDs to *peerIDs*.
+
+
+
+#### **[FS-A-PEER]**:
+Peers can be faulty, and we do not make any assumptions about the number or
+ratio of correct/faulty nodes. Faulty processes may be Byzantine
+according to [**[TMBC-AUTH-BYZ]**][TMBC-Auth-Byz-link].
+
+#### **[FS-A-VAL]**:
+The system satisfies [**[TMBC-AUTH-BYZ]**][TMBC-Auth-Byz-link] and [**[TMBC-FM-2THIRDS]**][TMBC-FM-2THIRDS-link]. Thus, there is a
+blockchain that satisfies the soundness requirements (that is, the
+validation rules in [[block]]).
+
+#### **[FS-A-COMM]**:
+Communication between the node *FS* and all correct peers is reliable and
+bounded in time: there is a message end-to-end delay *Delta* such that
+if a message is sent at time *t* by a correct process to a correct
+process, then it will be received and processed by time *t +
+Delta*. This implies that we need a timeout of at least *2 Delta* for
+remote procedure calls to ensure that the response of a correct peer
+arrives before the timeout expires.
+
+
+## Distributed Problem Statement
+
+### Design choices
+
+
+#### Two Kinds of Termination
+
+We do not assume that there is a correct full node in
+*peerIDs*. Under this assumption no protocol can guarantee the combination
+of the properties [FS-SEQ-LIVE] and
+[FS-SEQ-TERM] and [FS-SEQ-TERM-SYNC] described in the sequential
+specification above. Thus, in the (unreliable) distributed setting, we
+consider two kinds of termination (successful and failure) and we will
+specify below under what (favorable) conditions *Fastsync* ensures to
+terminate successfully, and satisfy the requirements of the sequential
+problem statement:
+
+#### **[FS-DISTR-TERM]**:
+*Fastsync* eventually terminates: it either *terminates successfully* or it  *terminates with failure*.
+
+
+#### Remote Functions
+
+The Tendermint Full Node exposes the following functions over
```

Comment:
We haven't defined what is a Tendermint full node. Can we just say that peers expose the following functions over RPC?